File: src/main/java/com/intgrah/algorithms/sort/InsertionSort.java
package com.intgrah.algorithms.sort;

import java.util.Comparator;

public class InsertionSort<T> extends Sorter<T> {

    public InsertionSort(Comparator<T> ord) {
        super(ord);
    }

    @Override
    public void sort(T[] a) {
        for (int j = 1; j < a.length; j++) {
            T x = a[j];
            int i = j;
            while (i > 0 && ord.compare(a[i - 1], x) > 0)
                a[i--] = a[i];
            a[i] = x;
        }
    }

}

File: sort/insertion_sort.py
def insertion_sort(a: list) -> None:
    for j in range(1, len(a)):
        x = a[j]
        i = j
        while i > 0 and a[i - 1] > a[i]:
            a[i] = a[i - 1]
            i -= 1
        a[i] = x
